File: baseble/src/main/java/com/fcbox/ble/baseble/exception/ConnectException.java
package com.fcbox.ble.baseble.exception;

import android.bluetooth.BluetoothGatt;

import com.fcbox.ble.baseble.common.BleExceptionCode;


/**
 * 连接异常
 */
@SuppressWarnings("unused")
public class ConnectException extends BleException {
    private BluetoothGatt bluetoothGatt;
    private int gattStatus;

    public ConnectException(BluetoothGatt bluetoothGatt, int gattStatus) {
        super(BleExceptionCode.CONNECT_ERR, "Connect Exception Occurred! ");
        this.bluetoothGatt = bluetoothGatt;
        this.gattStatus = gattStatus;
    }

    public int getGattStatus() {
        return gattStatus;
    }

    public ConnectException setGattStatus(int gattStatus) {
        this.gattStatus = gattStatus;
        return this;
    }

    public BluetoothGatt getBluetoothGatt() {
        return bluetoothGatt;
    }

    public ConnectException setBluetoothGatt(BluetoothGatt bluetoothGatt) {
        this.bluetoothGatt = bluetoothGatt;
        return this;
    }

    @Override
    public String toString() {
        return "ConnectException{" +
                "gattStatus=" + gattStatus +
                ", bluetoothGatt=" + bluetoothGatt +
                "} " + super.toString();
    }
}

File: baseble/src/main/java/com/xzy/ble/baseble/exception/ConnectException.kt
package com.xzy.ble.baseble.exception

import android.bluetooth.BluetoothGatt

import com.xzy.ble.baseble.common.BleExceptionCode


/**
 * 连接异常
 */
@Suppress("unused")
class ConnectException(private var bluetoothGatt: BluetoothGatt?, private var gattStatus: Int) : BleException(BleExceptionCode.CONNECT_ERR, "Connect Exception Occurred! ") {

    fun getGattStatus(): Int {
        return gattStatus
    }

    fun setGattStatus(gattStatus: Int): ConnectException {
        this.gattStatus = gattStatus
        return this
    }

    fun getBluetoothGatt(): BluetoothGatt? {
        return bluetoothGatt
    }

    fun setBluetoothGatt(bluetoothGatt: BluetoothGatt): ConnectException {
        this.bluetoothGatt = bluetoothGatt
        return this
    }

    override fun toString(): String {
        return "ConnectException{" +
                "gattStatus=" + gattStatus +
                ", bluetoothGatt=" + bluetoothGatt +
                "} " + super.toString()
    }
}
